File: archive/cold-storage/reference/aaes_os_v1/typescript/src/invariant_engine.ts
import type { GovernedSpan, InvariantId, TraceEvent } from "./types.js";
import type { InvariantEngine } from "./types.js";
import { notImplemented } from "./error.js";

export class InvariantEngineStub implements InvariantEngine {
  check(_event: TraceEvent, _span: GovernedSpan, _prior: TraceEvent[]) {
    return notImplemented<void>("InvariantEngine.check");
  }

  check_ids(
    _ids: InvariantId[],
    _event: TraceEvent,
    _span: GovernedSpan,
    _prior: TraceEvent[],
  ) {
    return notImplemented<void>("InvariantEngine.check_ids");
  }
}

File: archive/cold-storage/reference/aaes_os_v1/rust/src/invariant_engine.rs
use crate::error::{not_implemented, AaesResult};
use crate::governed_span::GovernedSpan;
use crate::types::{InvariantId, TraceEvent};

#[derive(Debug, Default)]
pub struct InvariantEngineStub;

impl InvariantEngineStub {
    pub fn check(
        &self,
        _event: &TraceEvent,
        _span: &GovernedSpan,
        _prior: &[TraceEvent],
    ) -> AaesResult<()> {
        Err(not_implemented("InvariantEngine.check"))
    }

    pub fn check_ids(
        &self,
        _ids: &[InvariantId],
        _event: &TraceEvent,
        _span: &GovernedSpan,
        _prior: &[TraceEvent],
    ) -> AaesResult<()> {
        Err(not_implemented("InvariantEngine.check_ids"))
    }
}
